apps/api/views/metrics.py

```diff
@@ -0,0 +1,761 @@
+import requests
+from copy import deepcopy
+
+from rest_framework.decorators import api_view, permission_classes
+from rest_framework_csv import renderers
+from rest_framework.response import Response
+from rest_framework.views import APIView
+from rest_framework import status, permissions
+
+from django.db.models import Count, F, Func, OuterRef, Max, Avg, Subquery
+from django.shortcuts import get_object_or_404
+
+from apps.api.serializers.metrics import InstructorMetricsSerializer, AdminMetricsSerializer
+
+from apps.profiles.models import StudentMembership, ChaUser, Instructor
+from apps.homework.models import Submission, Definition
+from apps.klasses.models import Klass
+from apps.groups.models import Team
+
+
+# To render with CSV, Run a list of dictionaries through a serializer with fields identical to the keys of the
+# renderer below. Then they'll be rendered into proper columns with labels (values associated with keys) at the
+# top row of the CSV document.
+
+class AdminSubmissionCSVRenderer(renderers.CSVRenderer):
+    labels = {
+        'date': 'Date',
+        'submissions_made_count': 'Submissions Made',
+        'submissions_total': 'Submissions Total',
+        'score_interval': 'Score Interval',
+        'score_interval_count': 'Scores on Interval',
+    }
+    header = list(labels.keys())
+
```

Comment:
Anyway we can make this a little more DRY? Seems the only thing we're changing for each is labels.